### ISSUE TITLE
Will Nvidia GPU work with wayfire?

### ISSUE THREAD

**Issue by pauledd** [WayfireWM/wayfire#1248]
Hi
I am not aware of the current situation. Will my Nvidia GPU GTX1070 work with wayfire?
I ask because there was some reports in the media recently stating that the new nvidia proprietary drivers
will now support wayland, what ever that means.

Can you give me a clarification?

**Comment by ammen99** [comment]
Hi, the problem is not Wayfire, but wlroots. Wayfire uses wlroots for the backend and regular OpenGL ES for drawing. I have also heard rumors that maybe nvidia will start to have support the same APIs as other drivers, which would mean that we'd get support for that automatically. But, I don't think nvidia has confirmed this officially, so I'm not holding my breath.

There is also a wlroots fork, https://github.com/danvd/wlroots-eglstreams. It aims to support the current proprietary drivers. I have not tested it, because I do not own any nvidia hardware. If you do, it would be interesting to try and compile that fork with Wayfire and report the results.

TLDR, official support may or may not happen, inofficial support hasn't been tested with Wayfire yet.

**Comment by pauledd** [comment]
Thank you for clarification.  I saw this fork and it seems to be quite buggy. Maybe I'll test that but keeping an eye on the wlroots issues for Nvidia related stuff.

**Comment by soreau** [comment]
It may be worth mentioning that wlroots compositors like wayfire work with the open source nouveau driver already, though it is a reverse engineered driver.

**Comment by ammen99** [comment]
> Free to join [the conversation](https://forum.garudalinux.org/t/garuda-wayfire-feedback/21812) @ammen99 @pauledd
> 
> My 2nd post outlines what is needed for proper dev ops.

New nvidia drivers should work with wlroots and Wayfire out of the box, though your comment is still valid for older GPUs which cannot be upgraded to the newest nvidia proprietary driver.
